### ISSUE TITLE
Wrong include orders ?

### ISSUE THREAD

**Issue by nkolban** [Samsung/libtuv#48]
The structure called struct uv__async is defined in uv__async.h

https://github.com/Samsung/libtuv/blob/bfc5089a415769d60e843b4359847adeca97c79a/include/uv__async.h#L80

however, in uv.h ... the include of uv_loop.h (which uses struct uv__async) occurs before the include of uv__async.h

**Comment by chokobole** [comment]
Yes, we will fix header include order soon! thanks for a good point!
